src/app/core/services/transaction-backend.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { AppConfig } from '../../app.config';
+import { Observable } from 'rxjs/Observable';
+import { map } from 'rxjs/operators';
+
+@Injectable()
+export class TransactionBackendService {
+    constructor(private http: HttpClient, private config: AppConfig) {
+    }
+
+    getUserTransaction(id: string | String): Observable<Transaction> {
+        return this.http
+            .get<Transaction>(`${this.getUrl()}/${id}`)
+    }
+
+    rollbackTransaction(id: string | String): Observable<any> {
+        return this.setTransactionAction(id, TransactionActions.ROLLBACK)
+    }
+
+    commitTransaction(id: string | String): Observable<any> {
+        return this.setTransactionAction(id, TransactionActions.COMMIT)
+    }
+
+    private setTransactionAction(id: string | String, action: TransactionActions) {
+        return this.http
+            .post<any>(`${this.getUrl()}/${id}/${action}`, {})
+            .pipe(map(data =>  data || []));
+    }
+
+    private getUrl() {
+        return `${this.config.getApiUrl()}/user-transaction`;
```

Comment:
Couldn't it be a variable set in the constructor?

Reply:
Nope because in constructor it is not yet availabile. The this.config gets populated after the first /config.json request happens which is far after constructor of this service is executed.